modules/cardinal_service_profile_helper/src/Controller/Documentation.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+namespace Drupal\cardinal_service_profile_helper\Controller;
+
+use Drupal\Core\Controller\ControllerBase;
+use Michelf\MarkdownExtra;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+/**
+ * Class Documentation.
+ *
+ * @package Drupal\cardinal_service_profile_helper\Controller
+ */
+class Documentation extends ControllerBase {
+
+  /**
+   * Controller callback to get the help document contents.
+   *
+   * @param string $document
+   *   Help file name.
+   *
+   * @return array
+   *   Help markdown markup converted to html.
+   */
+  public function getDocumentation($document) {
+    // Invalid path.
+    if (!file_exists(self::getProfilePath() . "/help/$document.md")) {
+      throw new NotFoundHttpException($this->t('Document @document not found', ['@document' => $document]));
+    }
+    $html = MarkdownExtra::defaultTransform(file_get_contents($this->getProfilePath() . "/help/$document.md"));
+    $h1 = preg_grep('/<h1>.*?<\/h1>/', explode("\n", $html));
+
+    return [
+      '#markup' => preg_replace('/<h1>.*?<\/h1>/', '', $html),
+      '#title' => !empty($h1[0]) ? strip_tags($h1[0]) : NULL,
+    ];
+  }
+
+  /**
+   * Page title callback.
+   *
+   * @param string $document
+   *   File name of the help document.
+   *
+   * @return string
+   *   Page title.
+   */
+  public function getTitle($document) {
+    $html = $this->getDocumentation($document);
+    return $html['#title'];
```

Comment:
Will there's always be a title or should there be a check for a condition?

Reply:
There should always be one, but null values don't throw errors anyways.

Reply:
Nevermind -- I see where this comes from now.